stdlib/urllib/parse.pyi

```diff
@@ -169,21 +193,27 @@ def urljoin(base: AnyStr, url: AnyStr | None, allow_fragments: bool = True) -> A
 def urlparse(url: str, scheme: str | None = "", allow_fragments: bool = True) -> ParseResult: ...
 @overload
 def urlparse(
-    url: bytes | bytearray | None, scheme: bytes | bytearray | None = ..., allow_fragments: bool = True
+    url: bytes | bytearray | None, scheme: bytes | bytearray | None, allow_fragments: bool = True
 ) -> ParseResultBytes: ...
 @overload
-def urlsplit(url: str, scheme: str | None = "", allow_fragments: bool = True) -> SplitResult: ...
+def urlparse(url: str | None, scheme: Literal[""] = "", allow_fragments: bool = True) -> ParseResult | ParseResultBytes: ...
```

Comment:
```suggestion
def urlparse(url: None, scheme: Literal[""] = "", allow_fragments: bool = True) -> ParseResultBytes: ...
```
Anything involving str is covered by the first overload, right?

Reply:
No, the first overload will only be selected if the value passed in is inferred to be of type `str` (or a subclass of `str`). If the value passed in is inferred to be of type `str | None`, the type checker will reject the first overload.

Similarly, the second overload covers passing in `None`, but if you have a value that's inferred to be of type `str | None`, the type checker will reject that overload when you pass in that value.

Reply:
Type checkers should be able to split unions. https://mypy-play.net/?mypy=latest&python=3.10&gist=3d5319a9d1060a7d95b1927bb8c67e9d

Reply:
Okay -- let's wait for mypy_primer to give its verdict on the current version, then I'll try this version :)

Reply:
You were right, I was wrong!